src/rgw/rgw_rest_swift.cc

```diff
@@ -92,6 +92,7 @@ static void dump_account_metadata(struct req_state * const s,
   dump_header(s, "X-Account-Object-Count", global_stats.objects_count);
   dump_header(s, "X-Account-Bytes-Used", global_stats.bytes_used);
   dump_header(s, "X-Account-Bytes-Used-Actual", global_stats.bytes_used_rounded);
+  dump_header(s, "Accept-Ranges", "bytes");
```

Comment:
The `Accept-Ranges` header is added by [`dump_content_length()`](https://github.com/ceph/ceph/blob/master/src/rgw/rgw_rest.cc#L448). By default the function isn't called for container and account listings to avoid extra buffering as they are both generated on-the-fly. For Swift this behavior can be changed with `rgw_swift_enforce_content_length`.

IIRC Tempest does check for `Content-Length` just after `Accept-Ranges`, so I'm not sure whether this patch changes much. Though, I'm not against merge if the header isn't duplicated in responses when `rgw_swift_enforce_content_length` is `true`. Could you please check this out?

Reply:
@rzarzynski Thanks for your comments. 
I re-run the Tempest test with the patch, there is no duplicated "Accept-Ranges" observed when I enable the option "rgw_swift_enforce_content_length" in ceph.conf. There is no difference between the Tempest test result when I enable or disable the option.

In OpenStack object storage tempest test, the "Accept-Ranges" is mandatory for container and account listing. By default the option "rgw_swift_enforce_content_length" is not set, the field  "Accept-Ranges" will not be set in the response head of Swift API. With the patch, some Tempest test failures will be fixed, and it can save some extra cost in Ceph.  What's your opinion?